Plugins/Reveal/Reveal.cpp

```diff
@@ -46,37 +46,40 @@ Reveal::Reveal(Services::ProxyServiceList* services)
 Reveal::~Reveal()
 {
 }
-int32_t Reveal::HookStealthDetection(CNWSCreature* pObserverCreature, CNWSCreature* pHidingCreature, int32_t bClearLOS, int32_t* bSeen, int32_t* bHeard, int32_t bTargetInvisible)
+BOOL Reveal::HookStealthDetection(CNWSCreature* pObserverCreature, CNWSCreature* pHidingCreature, BOOL bClearLOS, BOOL* bSeen, BOOL* bHeard, BOOL bTargetHiding)
 {
     if (pObserverCreature->m_bPlayerCharacter && pHidingCreature->m_bPlayerCharacter && pHidingCreature->m_nStealthMode)
     {
         if (pObserverCreature->GetArea() == pHidingCreature->GetArea())
         {
-            if (*pHidingCreature->nwnxGet<int>(revealKey + "PARTY"))
+            auto partyReveal = pHidingCreature->nwnxGet<int>(revealKey + "PARTY");
+            if (partyReveal && *partyReveal)
             {
                 if (pObserverCreature->GetFaction()->GetLeader() == pHidingCreature->GetFaction()->GetLeader())
                 {
-                    if(*pHidingCreature->nwnxGet<int>(detectionKey + "PARTY"))
-                    {
-                        *bSeen = true;
-                    }
+                    auto detectionVector = pHidingCreature->nwnxGet<int>(detectionKey + "PARTY");
+                    if (detectionVector && *detectionVector)
+                        *bSeen = *detectionVector;
```

Comment:
needs a

```
else
    *bSeen = false;
```

the variables that bSeen/bHeard are pointing to aren't initialized, so they should always be set to something

Reply:
Ah; done